mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py

```diff
@@ -0,0 +1,142 @@
+            inertial_euler = inertial.euler
+        if inertial.fullinertia is not None:
+            fullinertia = inertial.fullinertia
+        if inertial.diaginertia is not None:
+            fullinertia = [inertial.diaginertia[0],
+                           inertial.diaginertia[1],
+                           inertial.diaginertia[2],
+                           0,
+                           0,
+                           0]
+
+        # Mass is required, we don't need to check it
+        inertial = Inertiald(MassMatrix3d(inertial.mass,
+                                          Vector3d(fullinertia[0],
+                                                   fullinertia[1],
+                                                   fullinertia[2]),
+                                          Vector3d(fullinertia[3],
+                                                   fullinertia[4],
+                                                   fullinertia[5])),
+                             Pose3d(inertial_pos[0],
+                                    inertial_pos[1],
+                                    inertial_pos[2],
+                                    inertial_euler[0],
+                                    inertial_euler[1],
+                                    inertial_euler[2]))
+        link.set_inertial(inertial)
+
+    NUMBER_OF_VISUAL = 0
+    NUMBER_OF_COLLISION = 0
+
+    link.set_raw_pose(su.get_pose_from_mjcf(body))
+
+    for geom in body.geom:
+        # If the group is not defined then visual and collision is added
+        if geom.group is None:
+            visual = add_mjcf_visual_to_sdf(geom)
+            if visual is not None:
+                if geom.name is not None:
+                    visual.set_name("visual_" + geom.name)
+                else:
+                    visual.set_name("visual_" + str(NUMBER_OF_VISUAL))
+                    NUMBER_OF_VISUAL = NUMBER_OF_VISUAL + 1
+                visual.set_raw_pose(su.get_pose_from_mjcf(geom))
+                link.add_visual(visual)
+
+            col = add_mjcf_collision_to_sdf(geom)
+            if col is not None:
+                if geom.name is not None:
+                    col.set_name("collision_" + geom.name)
+                else:
+                    col.set_name("collision_" + str(NUMBER_OF_COLLISION))
+                    NUMBER_OF_COLLISION = NUMBER_OF_COLLISION + 1
+                col.set_raw_pose(su.get_pose_from_mjcf(geom))
+                link.add_collision(col)
+        elif geom.group == VISUAL_GEOM_GROUP:
+            visual = add_mjcf_visual_to_sdf(geom)
+            if visual is not None:
+                if geom.name is not None:
+                    visual.set_name("visual_" + geom.name)
+                else:
+                    visual.set_name("visual_" + str(NUMBER_OF_VISUAL))
+                    NUMBER_OF_VISUAL = NUMBER_OF_VISUAL + 1
+                visual.set_raw_pose(su.get_pose_from_mjcf(geom))
+                link.add_visual(visual)
```

Comment:
This is repeated. Consider refactoring into a function.

Reply:
https://github.com/gazebosim/gz-mujoco/pull/26/commits/2eb7da53bd3c9ef6168e09191a61b0975a6f5dbb